### PR TITLE
Provide a better exception message.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
You cannot call VSWorkspace.OpenDocument on a background thread.   We now give an up front exception to make the issue clear to the caller as to what the issue is (instead of a cryptic 'Assert-False' error further on down the line).

Fixes https://github.com/dotnet/roslyn/issues/4613

(well, doesn't actually fix.  just improves the API experience for people).

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/mlangide

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs [hunk 1]
```diff
@@ -672,6 +675,11 @@ public void OpenDocumentCore(DocumentId documentId, bool activate = true)
                 throw new ArgumentNullException(nameof(documentId));
             }
 
+            if (!foregroundObject.IsForeground())
+            {
+                throw new NotSupportedException(ServicesVSResources.ThisWorkspaceOnlySupportsOpeningDocumentsOnTheUIThread);
```
Comment (tmeschter) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs]: Should this be a `NotSupportedException` or an `InvalidOperationException`?

Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs]: I think InvalidOperationException would be better.  I'll switch that.  Thanks!

**Comment by dpoeschl** [comment]
:+1:

**Pull request merged**